apps/taiga-lumbermill/src/dashboards/theme-generator/theme-generator.component.ts

```diff
@@ -26,11 +37,51 @@ import {data} from './theme-generator.constants';
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class ThemeGeneratorComponent {
+    private readonly clipboard = inject(Clipboard);
+    private readonly activatedRoute = inject(ActivatedRoute);
+    protected readonly url = location.href;
+    protected params = toSignal(this.activatedRoute.queryParams)();
     protected themeData = data;
     protected readonly palette = TUI_DEFAULT_INPUT_COLORS;
-    protected colors = this.themeData.map((val) => signal(val.initialValue));
+    protected colors = this.themeData.map((val) => {
+        const result = this.params?.[val.variable] ?? val.initialValue;
+
+        return signal((result[0] !== 'r' && result[0] !== '#' ? '#' : '') + result);
+    });
+
+    protected copied = signal(false);
+    protected shared = signal(false);
 
     protected theme = computed(() =>
         this.colors.map((val, i) => `${this.themeData[i].variable}: ${val()};`).join(' '),
     );
+
+    protected copy(): void {
+        this.copied.set(true);
+        setTimeout(() => {
+            this.copied.set(false);
+        }, 800);
+        let text = ':root {';
+
+        for (let i = 0; i < this.themeData.length; i++) {
+            text += `\n${this.themeData[i].variable}: ${this.colors[i]()};`;
+        }
+
+        text += '\n}';
+        this.clipboard.copy(text);
+    }
+
+    protected share(): void {
```

Comment:
This doesn't work. Nothing happened after going on the link with the theme in the URL